internal/network/id.go

```diff
@@ -0,0 +1,36 @@
+package network
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/oklog/ulid"
+)
+
+var _ ID = (*id)(nil)
+
+type id ulid.ULID
+
+var (
+	lock       sync.Mutex
+	randSource = rand.New(rand.NewSource(time.Now().UnixNano()))
+	entropy    = ulid.Monotonic(randSource, 0)
+)
+
+func createID() ID {
+	lock.Lock()
+	defer lock.Unlock()
+
+	id, err := ulid.New(ulid.Timestamp(time.Now()), entropy)
+	if err != nil {
+		log.Fatal(fmt.Errorf("new ulid: %w", err))
```

Comment:
Will this stop the execution, since its a "fatal" ? 
Will it really matter if this stops the execution?

Reply:
Yes it will, this has to be changed

Reply:
I documented a lot, I don't know how I could remove that, and I am also not sure if we really need to. If this breaks, that implies that something is really really broken, and a fallback would be, to just increment a number atomically and use that as an ID, which would be a quick and reliable fix.